File: aoc2019/aoc2019-java/src/test/java/de/havox_design/aoc2019/day02/Day02Test.java
package de.havox_design.aoc2019.day02;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day02Test {

    @ParameterizedTest
    @MethodSource("getDataForTask1")
    void testTask1(String fileName, long expectation) {
        Assertions.assertEquals(expectation, ProgramAlarm1202.processTask1(fileName));
    }

    private static Stream<Arguments> getDataForTask1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day02/day02part1sample1.txt", 2L),
                Arguments.of("de/havox_design/aoc2019/day02/day02part1sample2.txt", 2L),
                Arguments.of("de/havox_design/aoc2019/day02/day02part1sample3.txt", 2L),
                Arguments.of("de/havox_design/aoc2019/day02/day02part1sample4.txt", 30L)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForTask2")
    void testTask2(String fileName, long expectation) {
        Assertions.assertEquals(expectation, ProgramAlarm1202.processTask2(fileName));
    }

    private static Stream<Arguments> getDataForTask2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day02/day02part2sample.txt", 0L)
        );
    }
}

File: aoc2020/aoc2020-kotlin/src/test/kotlin/de/havox_design/aoc2020/day02/Day02Test.kt
package de.havox_design.aoc2020.day02

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day02Test {

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart1")
    fun testProcessPart1(filename: String, expectedResult: Int) =
        PasswordPhilosophy(filename).processPart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart2")
    fun testProcessPart2(filename: String, expectedResult: Long) =
        PasswordPhilosophy(filename).processPart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestProcessPart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2020/day02/day02part1sample1.txt", 2),
                Arguments.of("de/havox_design/aoc2020/day02/day02part1sample2.txt", 1),
                Arguments.of("de/havox_design/aoc2020/day02/day02part1sample3.txt", 0),
                Arguments.of("de/havox_design/aoc2020/day02/day02part1sample4.txt", 1)
            )

        @JvmStatic
        private fun getDataForTestProcessPart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2020/day02/day02part2sample.txt", 0L)
            )
    }
}
